tests/pytest/test_utils.py

```diff
@@ -36,3 +38,40 @@ def test_str_deprecated(dut):
 
         diff_bytes = cocotb.utils.hexdiffs(b'\x20\x65\x00\xff', b'\x20\x00\x65')
         assert diff_bytes == diff_str
+
+
+def test_pack_deprecated():
+
+    class Example(ctypes.Structure):
+        _fields_ = [
+            ("a", ctypes.c_byte),
+            ("b", ctypes.c_byte)]
+
+    e = Example(a=0xCC, b=0x55)
+
+    with pytest.warns(DeprecationWarning):
+        a = cocotb.utils.pack(e)
+
+    assert a == bytes(e)
+
+
+def test_unpack_deprecated():
+
+    class Example(ctypes.Structure):
+        _fields_ = [
+            ("a", ctypes.c_byte),
+            ("b", ctypes.c_byte)]
+
+    e = Example(a=0xCC, b=0x55)
+    f = Example(a=0xCC, b=0x55)
+
+    b = b'\x01\x02'
+
+    with pytest.warns(DeprecationWarning):
+        cocotb.utils.unpack(e, b)
+
+    assert e.a == 1 and e.b == 2
+
+    memoryview(f).cast('B')[:] = b
```

Comment:
Changed this line, previously we were testing `ctype.memmove` instead of the actual recommended solution. Will merge if CI is happy.